src/HIE/Bios/Cradle.hs

```diff
@@ -457,13 +456,14 @@ cabalAction work_dir mc l fp = do
 removeInteractive :: [String] -> [String]
 removeInteractive = filter (/= "--interactive")
 
--- Strip out any ["+RTS", ..., "-RTS"] sequences in the command string list.
+-- | Strip out any ["+RTS", ..., "-RTS"] sequences in the command string list.
 removeRTS :: [String] -> [String]
```

Comment:
```suggestion
data InRTS = OutsideRTS | InsideRTS

removeRTS :: [String] -> [String]
removeRTS = go OutsideRTS where
  go :: InRTS -> [String] -> [String]
  go _ [] = []
  go OutsideRTS (y:ys)
    | "+RTS" `isPrefixOf` y = removeRTS (if "-RTS" `isSuffixOf y then OutsideRTS else InsideRTS) ys
    | otherwise = y : removeRTS ys
  go InsideRTS (y:ys) = removeRTS (if y = "-RTS" then OutsideRTS else InsideRTS) ys
```